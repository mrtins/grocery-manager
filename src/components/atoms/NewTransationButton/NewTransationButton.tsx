import classNames from 'classnames';
import { HiPlus } from 'react-icons/hi';

interface NewTransationButtonProps {
  collapseSidebar: boolean;
  handleClick: () => void;
}

export default function NewTransationButton({
  collapseSidebar,
  handleClick,
}: NewTransationButtonProps) {
  return (
    <button
      type="button"
      onClick={handleClick}
      className={classNames(
        'new-transaction-button flex',
        `${!collapseSidebar ? 'w-36 justify-evenly items-center' : 'mb-6'} `,
      )}
    >
      <HiPlus /> <span>{!collapseSidebar && 'Novo'}</span>
    </button>
  );
}
