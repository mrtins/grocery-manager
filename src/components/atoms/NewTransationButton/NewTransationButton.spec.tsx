import { render, screen, fireEvent } from '@testing-library/react';
import NewTransationButton from './NewTransationButton';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

describe('NewTransationButton component', () => {
  it('should render button with correct CSS classes when collapseSidebar is falsy', () => {
    render(
      <NewTransationButton collapseSidebar={false} handleClick={() => null} />,
    );

    const button = screen.getByRole('button');

    expect(button.classList.toString()).toEqual(
      'new-transaction-button flex w-36 justify-evenly items-center ',
    );
  });

  it('should render button with correct CSS classes when collapseSidebar is truthy', () => {
    render(
      <NewTransationButton collapseSidebar={true} handleClick={() => null} />,
    );

    const button = screen.getByRole('button');

    expect(button.classList.toString()).toEqual(
      'new-transaction-button flex mb-6 ',
    );
  });

  it('should render NewTransationButton with button title when collapseSidebar prop is falsy', () => {
    render(
      <NewTransationButton collapseSidebar={false} handleClick={() => null} />,
    );

    expect(screen.getByText('Novo')).toBeInTheDocument();
  });

  it('should return null on button click', () => {
    render(
      <NewTransationButton collapseSidebar={false} handleClick={() => null} />,
    );

    const button = screen.getByRole('button');

    expect(button).toMatchSnapshot();
  });
});
