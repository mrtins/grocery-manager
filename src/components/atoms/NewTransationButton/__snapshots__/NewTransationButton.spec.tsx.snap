// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTransationButton component should return null on button click 1`] = `
<button
  class="new-transaction-button flex w-36 justify-evenly items-center "
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 20 20"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
      fill-rule="evenodd"
    />
  </svg>
   
  <span>
    Novo
  </span>
</button>
`;
